pkg/service/annotation.go

```diff
@@ -0,0 +1,52 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/pyroscope-io/pyroscope/pkg/model"
+	"gorm.io/gorm"
+)
+
+type AnnotationsService struct{ db *gorm.DB }
+
+func NewAnnotationsService(db *gorm.DB) AnnotationsService {
+	return AnnotationsService{db: db}
+}
+
+// CreateAnnotation creates a single annotation for a given application
+// It does not check if the application has consumed any data
+func (svc AnnotationsService) CreateAnnotation(ctx context.Context, params model.CreateAnnotation) (*model.Annotation, error) {
+	var u model.Annotation
+
+	if err := params.Validate(); err != nil {
+		return nil, err
+	}
+
+	u.AppName = params.AppName
+	u.Content = params.Content
+	u.Timestamp = params.Timestamp
+
+	tx := svc.db.WithContext(ctx)
+
+	// Upsert
+	if err := tx.Where(model.CreateAnnotation{
```

Comment:
Although it's simple enough, it already has a certain logic (upserting). 

@kolesnikovae What do you think about adding yet another indirection layer with the `Repository` pattern?

Reply:
>I'd better re-implement services for that alternative database/storage
Alright!

Although we would need to figure out where to put that common business logici independent of storage